### PR TITLE
Index posting list

### PR DESCRIPTION
**Pull request by hanahmily**
The changes introduce index posting list as

 1. the search result of the index module
 2. the input of `FetchEntities` 

The posting list replaces `common.ChunkIDs` as the canonical collection of `common.ChunkID`.

### CODE REVIEW

File: pkg/query/logical/plan_indexscan.go [hunk 1]
```diff
@@ -107,12 +108,13 @@ type indexScan struct {
...
 func (i *indexScan) Execute(ec executor2.ExecutionContext) ([]data.Entity, error) {
-	var chunkSet common.ChunkIDs
+	var chunkSet posting.List
 	for _, exprs := range i.conditionMap {
 		// TODO: Discuss which metadata should be used!
 		// 1) traceSeries Metadata: indirect mapping
 		// 2) indexRule Metadata: cannot uniquely determine traceSeries
-		chunks, err := ec.Search(*i.traceMetadata, i.startTime, i.endTime, convertToConditions(exprs))
+		// TODO: should pass correct shardID
+		chunks, err := ec.Search(*i.traceMetadata, 0, i.startTime, i.endTime, convertToConditions(exprs))
```
Comment (hanahmily) [inline pkg/query/logical/plan_indexscan.go:117]: @lujiajing1126 You should get shards info from the series module, then searching results among those shards.

File: pkg/query/logical/plan_indexscan.go [hunk 2]
```diff
@@ -121,11 +123,12 @@ func (i *indexScan) Execute(ec executor2.ExecutionContext) ([]data.Entity, error
...
 		} else {
 			// afterwards, it must not be nil
-			chunkSet = chunkSet.HashIntersect(chunks)
+			_ = chunkSet.Intersect(chunks)
 		}
 	}
 
-	return ec.FetchEntity(*i.traceMetadata, chunkSet, series.ScanOptions{
+	//TODO: pass correct shardID
+	return ec.FetchEntity(*i.traceMetadata, 0, chunkSet, series.ScanOptions{
```
Comment (hanahmily) [inline pkg/query/logical/plan_indexscan.go:131]: @lujiajing1126 As mentioned above, once getting the searching results from a particular shard, you would fetch entities from that shard's trace series.

File: pkg/query/logical/plan_indexscan.go [hunk 3]
```diff
@@ -107,12 +108,13 @@ type indexScan struct {
 }
 
 func (i *indexScan) Execute(ec executor2.ExecutionContext) ([]data.Entity, error) {
-	var chunkSet common.ChunkIDs
+	var chunkSet posting.List
 	for _, exprs := range i.conditionMap {
 		// TODO: Discuss which metadata should be used!
```
Comment (hanahmily) [inline pkg/query/logical/plan_indexscan.go:113]: @lujiajing1126 IndexRule metadata should be selected here

Comment (lujiajing1126) [reply pkg/query/logical/plan_indexscan.go:113]: But as I understand, a IndexRule probably can be bounded to several Series. So there is a 1:N relationship between IndexRule and Series

Comment (hanahmily) [reply pkg/query/logical/plan_indexscan.go:113]: Sorry, it should be the `RuleBinding` which represents the relationship between `IndexRule` and `Series`

**Pull request merged**